### PR TITLE
OLI API login via access key

### PR DESCRIPTION
**Pull request by veccp**
## Fixes/Resolves:

Simplifies logins to OLI Cloud API

## Summary/Motivation:

Required to improve OLI Cloud API test coverage

## Changes proposed in this PR:
- Create/save access keys
- Delete access keys

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by veccp** [comment]
@lbianchi-lbl I think we're ready to look at adding the access key to tests

File: watertap/tools/oli_api/tests/test_credentials.py [hunk 1]
```diff
@@ -8,59 +8,95 @@
...
+    test_contents = listdir(test_dir)
+
+    credential_manager.login()
+    with OLIApi(credential_manager, test=True) as oliapi:
+        local_dbs_file = join(test_dir, "test.dbs")
+        oliapi.get_dbs_file_id(local_dbs_file)
+        yield oliapi
+
+    _cleanup(root_contents, root_dir)
+    _cleanup(test_contents, test_dir)
```
Comment (lbianchi-lbl) [inline watertap/tools/oli_api/tests/test_credentials.py]: Since it's not being used in this test file and we have the fixture defined in `conftest.py`, can this be deleted?

**Comment by bknueven** [comment]
It looks like some tests are being skipped still I would not expect:
Screenshot 2023-12-11 at 12 44 39 PM

**Comment by lbianchi-lbl** [comment]
> It looks like some tests are being skipped still I would not expect: Screenshot 2023-12-11 at 12 44 39 PM

The tests that are being skipped are the ones that use the `auth_credentials` fixture, including those tests that use it indirectly through the derived `oliapi_instance` fixture. The fixture is set up to look for credentials in environment variables; if the variables are not found, `pytest.skip()` is called, which will skip any tests that require that fixture without having to implement the skip logic in the test itself. If all of the `OLI_API_XYZ` variables are found in the environment, the tests are not skipped:

https://github.com/watertap-org/watertap/blob/e4d53d8e5d3ca68adfb6b87474dc634fc2c38949/watertap/tools/oli_api/conftest.py#L69-L74

In summary: I'm not sure whether the above fully addresses your comment, but the fact that those tests are being skipped here (i.e. in the `watertap-org/watertap` CI) is by design and expected.

File: watertap/tools/oli_api/tests/test_credentials.py (hunk 1 above)
Comment (veccp) [reply watertap/tools/oli_api/tests/test_credentials.py]: Yes, will push my changes to the test.credentials file shortly

**Comment by bknueven** [comment]
> In summary: I'm not sure whether the above fully addresses your comment, but the fact that those tests are being skipped here (i.e. in the `watertap-org/watertap` CI) is by design and expected.

Right, of course.

**Review by lbianchi-lbl — APPROVED** [review]
The missing coverage is expected, with progress to address this being tracked in #1249.

**Review by bknueven — APPROVED** [review]
